File: src/image.c
#include "include/image.h"

void invert_colors(BITMAP* bmp)
{
    uint32_t image_size = (bmp->header.width * bmp->header.height);

    for (uint32_t i = 0; i < image_size; i++)
    {
        RGB* pixel = &bmp->pixels[i];

        pixel->b = 255 - pixel->b;
        pixel->g = 255 - pixel->g;
        pixel->r = 255 - pixel->r;
    }

    return;
}

File: src/image.rs
use crate::bitmap;

pub fn invert_colors(bmp: &mut bitmap::BITMAP) {
    for i in 0..(bmp.header.width * bmp.header.height) {
        let original_pixel = bmp.pixels.get(i as usize).unwrap();

        let inverted_pixel: bitmap::RGB = bitmap::RGB {
            r: 255 - original_pixel.r,
            g: 255 - original_pixel.g,
            b: 255 - original_pixel.b
        };

        bmp.pixels[i as usize] = inverted_pixel;
    }
}